src/main/java/no/unit/nva/doi/fetch/DoiValidator.java

```diff
@@ -0,0 +1,49 @@
+package no.unit.nva.doi.fetch;
+
+import java.net.URL;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.Objects.isNull;
+
+public final class DoiValidator {
+
+    public static final Pattern DOI_URL_PATTERN =
```

Comment:
Regex for doi? I think the only valid way of testing is to run a head request on the doi itself (i.e. not crossref/datacte, but the doi URI)

Reply:
Created new task: https://unit.atlassian.net/browse/NP-854